### PR TITLE
issue 3044 - use correct TransformerFactory impl class

### LINKED ISSUE #3044: issue-2266 implementation causes regression of 1226 FHIRGeneratorException when serializing resource <div> element as XML
**Describe the bug**
Issue #1226 was implemented to ensure that fhir-model always uses the correct TransformerFactory class.

Issue #2266 changed the implementation to address an illegal reflective action warning which would become an error in a future version of Java. However, this causes a regression in behavior in one case where the user is using fhir-validation-dist, not the whole server.

**Environment**
Which version of IBM FHIR Server: 4.10.0 and 4.10.1.

**To Reproduce**
1. See #1226 

**Expected behavior**
The test described by #1226 should pass.

**Additional context**

### PR DESCRIPTION
**Pull request by punktilious**
Signed-off-by: Robin Arnold <robin.arnold@ibm.com>

Fixes #3044.

### CODE REVIEW

**Review by lmsurpre — APPROVED** [review]
I had tried leaving this configurable per the design of these XML factories, but I agree it makes more sense just to instantiate a known-good impl since we know there are issues with other implementations.  LGTM.

File: fhir-model/src/main/java/com/ibm/fhir/model/util/XMLSupport.java [hunk 1]
```diff
@@ -226,15 +226,8 @@ private static XMLOutputFactory createXMLOutputFactory() {
 
     private static TransformerFactory createTransformerFactory() {
         try {
-            boolean isSet = System.getProperty(PROP_TRANSFORMER_FACTORY) != null;
-            if (!isSet) {
-                System.setProperty(PROP_TRANSFORMER_FACTORY, TRANSFORMER_FACTORY_IMPL);
```
Comment (lmsurpre) [inline fhir-model/src/main/java/com/ibm/fhir/model/util/XMLSupport.java:231]: we can probably remove the PROP_TRANSFORMER_FACTORY now that its not used here

File: fhir-persistence-schema/README.md [hunk 2]
```diff
@@ -493,3 +493,282 @@ If there is data in the DOMAINRESOURCE and RESOURCE table groups, which is unexp
 
 
 FHIR® is the registered trademark of HL7 and is used with the permission of HL7.
+
+
+# Physical Data Model
```
Comment (lmsurpre) [inline fhir-persistence-schema/README.md:498]: looks like some great doc!  did you mean to include it in this PR?

**Review by lmsurpre — APPROVED** [review]
LGTM

File: fhir-model/src/main/java/com/ibm/fhir/model/util/XMLSupport.java (hunk 1 above)
Comment (punktilious) [reply fhir-model/src/main/java/com/ibm/fhir/model/util/XMLSupport.java:231]: Ah yes...copied it back when I reverted the isSet stuff. I'll take it out

Comment (punktilious) [reply fhir-model/src/main/java/com/ibm/fhir/model/util/XMLSupport.java:231]: Wait...no...it's not there - already gone

**Pull request merged**